src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringFromThis = '';
```

Comment:
Bad variable name, it should describe what is stored in this variable (result of `join` method)
Also better to avoid using data type in variable name